health-integrations/src/main/java/com/netflix/runtime/health/status/ArchaiusHealthStatusFilterModule.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.runtime.health.status;
+
+import javax.inject.Singleton;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.netflix.archaius.ConfigProxyFactory;
+import com.netflix.runtime.health.api.HealthIndicator;
+import com.netflix.runtime.health.api.IndicatorFilter;
+import com.netflix.runtime.health.api.IndicatorFilters;
+import com.netflix.runtime.health.eureka.EurekaHealthStatusBridgeModule;
+import com.netflix.runtime.health.servlet.HealthStatusServlet;
+
+/***
+ * Installing this module will provide an implementation of {@link IndicatorFilter} backed by Archaius2. 
+ * This {@link IndicatorFilter} will be used by both the {@link EurekaHealthStatusBridgeModule} and
+ * the {@link HealthStatusServlet} to specify which {@link HealthIndicator}s will be used in 
+ * determining the healthiness of your application. 
+ */
+public class ArchaiusHealthStatusFilterModule extends AbstractModule {
+
+    @Override
+    protected void configure() {
+    }
+
+    @Provides
+    @Singleton
+    public HealthStatusInclusionConfiguration healthConfiguration(ConfigProxyFactory factory) {
+        return factory.newProxy(HealthStatusInclusionConfiguration.class);
+    }
+    
+    @Provides
+    @Singleton
+    public IndicatorFilter indicatorFilter(HealthStatusInclusionConfiguration config) {
+        return new ArchaiusDrivenStatusFilter(config);
+    }
+
+    private static class ArchaiusDrivenStatusFilter implements IndicatorFilter {
+
+        private final HealthStatusInclusionConfiguration config;
+
+        public ArchaiusDrivenStatusFilter(HealthStatusInclusionConfiguration config) {
+            this.config = config;
+        }
+
+        @Override
+        public boolean matches(String indicatorName) {
+            return IndicatorFilters
+                    .includes(config.includedIndicators())
+                    .excludes(config.excludedIndicators()).build()
```

Comment:
We are rebuilding every time. Talked to David about using the property change listener events and synchronizing on some internal reference to that property, but it didn't seem to be worth it given the (in)frequency of these calls. As for changes to config in the middle of an operation, once we get past the creation of this filter it isn't an issue. It is possible that includes and excludes could be updated simultaneously but read independently. But there is no api for transactions around config changes. Worst case, we have the update for only one of the two properties for a single iteration of the health check.